scripts/split-e2e-tests.ts

```diff
@@ -380,6 +381,37 @@ function verifyConfig() {
 
 function main(): void {
   const config = loadConfig();
+  
+  // make sure that only valid paths are used to store artifacts/results
+  const storagePathsUsedInConfig = new Set();
+  const invalidPaths = new Set();
+  for(let key of Object.keys(config.jobs)) {
+    const job = config.jobs[key];
+    const steps = job.steps;
+    
+    for(let i = 0; i < steps.length; i ++){
```

Comment:
can we use `for (const step of steps)`

Reply:
steps is a custom type, and doesn't implement the iterator interface unfortunately